tools/check-content.sh

```diff
@@ -1,17 +1,33 @@
-#!/bin/sh -e
```

Comment:
For all the shell script changes:

- Not sure why it's desirable to have different exit statuses depending on whether the script is running in CI or not.

- The cosmetic changes are good per se but they make the script more complex. I'd rather keep these shell scripts as simple as possible.

Reply:
> Not sure why it's desirable to have different exit statuses depending on whether the script is running in CI or not.

This allows `make lint` when run interactively to execute all checks rather than short-circuiting when a particular check fails. In an interactive context the developer is going to notice if there is a problem by observing the error messages. Especially with the other changes to only produce output from a test if it detects a problem. This means the developer will need to run `make lint` fewer times to resolve all problems. In the CI environment you want the detection of a problem to be noticed via the exit status. This refinement isn't, strictly speaking, necessary so I'll just remove that logic.

> The cosmetic changes are good per se but they make the script more complex. I'd rather keep these shell scripts as simple as possible.

I would argue they do not increase the complexity (especially if the exit status refinement is removed). Those changes are also not merely cosmetic. They greatly improve the usability by only producing output if a problem is detected. This makes it significantly easier to look at the `make lint` output and determine, at a glance, whether or not a problem was found.